File: code/library/src/main/java/org/fbme/lib/iec61499/declarations/ParameterAssignment.java
package org.fbme.lib.iec61499.declarations;

import org.fbme.lib.common.ContainedElement;
import org.fbme.lib.common.Declaration;
import org.fbme.lib.common.Reference;
import org.fbme.lib.st.expressions.Literal;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface ParameterAssignment extends ContainedElement {

    @Nullable Declaration getContainer();

    @NotNull Reference<ParameterDeclaration> getParameterReference();

    @Nullable Literal<?> getValue();

    void setValue(@Nullable Literal<?> value);
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/declarations/ParameterAssignment.kt
package org.fbme.lib.iec61499.declarations

import org.fbme.lib.common.ContainedElement
import org.fbme.lib.common.Declaration
import org.fbme.lib.common.Reference
import org.fbme.lib.st.expressions.ParameterValue

interface ParameterAssignment : ContainedElement {
    override val container: Declaration?
    val parameterReference: Reference<ParameterDeclaration>
    var value: ParameterValue?
}